### ISSUE TITLE
no-unused-components doesn't detect $options.components.Component addressing

### ISSUE THREAD

**Issue by Jessidhia** [vuejs/eslint-plugin-vue#778]
**Tell us about your environment**	
* **ESLint version:**	5.12.0
* **eslint-plugin-vue version:**	5.1.0
* **Node version:**	11.0.0

**Please show your full configuration:**

```js
module.exports = {
  extends: [
    'plugin:vue/recommended'
  ],
  rules: {
    'vue/name-property-casing': ['error', 'kebab-case'],
    'vue/html-indent': ['error', 2, {
      'attribute': 1,
      'baseIndent': 1,
      'closeBracket': 0,
      'alignAttributesVertically': false,
      'ignores': []
    }],
    'vue/require-default-prop': 'off',
    'vue/max-attributes-per-line': 'off',
    'vue/no-v-html': 'off'
  }
};
```

**What did you do?**	

```vue
<template>
  <foo-comp
    :attr="$options.components.BarComp"
  />
</template>

<script>
import { FooComp, BarComp } from './components'

export default {
  components: { FooComp, BarComp }
}
</script>
```

**What did you expect to happen?**
`BarComp` to be considered as used, as it's directly accessed through `$options.components`

**What actually happened?**
`The "BarComp" component has been registered but not used.`

---

This might be a discouraged pattern, but I don't know what would be the correct pattern to replace this with if that's the case.

This is similar, but not quite the same, as the `<component :is/>` case. The component's name is still present directly in the template.

**Comment by ota-meshi** [comment]
Hello.
I think that the problem will be solved if you change as follows.

```vue
<template>
  <foo-comp
    :attr="BarComp"
  />
</template>

<script>
import { FooComp, BarComp } from './components'

export default {
  components: { FooComp },
  data() {
    return { BarComp }
  }
}
</script>
```

**Comment by ota-meshi** [comment]
If your issue persists please reopen this issue.
